enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/components/SdkIframeEmbedRoute.tsx

```diff
@@ -0,0 +1,86 @@
+import { useMemo } from "react";
+import { P, match } from "ts-pattern";
+
+import {
+  InteractiveDashboard,
+  InteractiveQuestion,
+  defineMetabaseAuthConfig,
+} from "embedding-sdk";
+import { MetabaseProvider } from "embedding-sdk/components/public/MetabaseProvider";
+import { Box } from "metabase/ui";
+import { hasPremiumFeature } from "metabase-enterprise/settings";
+
+import { useSdkIframeEmbedEventBus } from "../hooks/use-sdk-iframe-embed-event-bus";
+import type { SdkIframeEmbedSettings } from "../types/embed";
+
+import { SdkIframeInvalidLicenseError } from "./SdkIframeError";
+
+export const SdkIframeEmbedRoute = () => {
+  const { embedSettings } = useSdkIframeEmbedEventBus();
+
+  const authConfig = useMemo(() => {
+    if (!embedSettings) {
+      return;
+    }
+
+    // TODO: add support for SSO auth once the new SSO implementation on the SDK is ready
+    if (!embedSettings.instanceUrl || !embedSettings.apiKey) {
+      return;
+    }
+
+    return defineMetabaseAuthConfig({
+      metabaseInstanceUrl: embedSettings.instanceUrl,
+      apiKey: embedSettings.apiKey,
+    });
+  }, [embedSettings]);
+
+  // The SDK will show its own loading indicator, so we don't need to show it twice.
+  if (embedSettings === null || !authConfig) {
+    return null;
+  }
+
+  const hasEmbedTokenFeature = hasPremiumFeature("embedding_iframe_sdk");
+
+  // If the parent page is not running on localhost and
+  // the token feature is not present, we show an error message
+  if (!embedSettings._isLocalhost && !hasEmbedTokenFeature) {
+    return <SdkIframeInvalidLicenseError />;
+  }
```

Comment:
We want to let folks without licenses play with the SDK in localhost. We check the origin of the embedder instead of the embed frame's origin, because most people will likely try out new iframe embedding with Cloud starter.